torch/_inductor/codegen/cpp.py

```diff
@@ -2589,24 +2589,28 @@ def reduction_combine_vec(
     def indirect_assert(self, var, lower, upper, mask=None):
         assert isinstance(var, CppCSEVariable)
         assert var.dtype is not None
-        if not var.is_vec:
+        if not (var.is_vec or (mask and mask.is_vec)):
             return super().indirect_assert(var, lower, upper, mask)
+        if var.is_vec:
+            var_str = str(var)
+        else:
+            var_str = f"{self._get_vec_type(var.dtype)}({var})"
         lower_scalar = lower
         upper_scalar = upper
         if lower:
             lower = f"{self._get_vec_type(var.dtype)}({lower})"
         if upper:
             upper = f"{self._get_vec_type(var.dtype)}({upper})"
         if lower and upper:
-            cond = f"({lower} <= {var}) & ({var} < {upper})"
-            cond_print = f"{lower_scalar} <= {var} < {upper_scalar}"
+            cond = f"({lower} <= {var_str}) & ({var_str} < {upper})"
+            cond_print = f"{lower_scalar} <= {var_str} < {upper_scalar}"
```

Comment:
for the print we want to print just `var` for clarity. This becomes moot when the comment above is acted on though.